### PR TITLE
close #244

### LINKED ISSUE #244: mouse zooming at the edges
in the candelstick demo, if you use mouse to zoom, sometimes the edge ticks would be excluded or missed. or if you hold the mouse left button and swing the mouse out of canvas, the selection range could be random. ideally, when mouse cursor is out of range, every ticks along that direction should be included?

### PR DESCRIPTION
**Pull request by EmiPhil**
unless we want proxthresh to be a setting #244

### CODE REVIEW

File: src/uPlot.js [hunk 1]
```diff
@@ -1868,11 +1868,55 @@ export default function uPlot(opts, data, then) {
...
+			let _dragging = dragging;
+
+			if (dragging) {
+				// handle case when mousemove aren't fired all the way to edges by browser
+				let dLft = mouseLeft1;
+				let dRgt = plotWidCss - mouseLeft1;
+				let dTop = mouseTop1;
+				let dBtm = plotHgtCss - mouseTop1;
+
+				let xProx = true;
```
Comment (leeoniya) [inline src/uPlot.js:1881]: if these are booleans, let's use `snap` instead of `prox`. e.g. `snapX`.

File: src/uPlot.js [hunk 2]
```diff
@@ -1868,11 +1868,55 @@ export default function uPlot(opts, data, then) {
...
+				let yProx = true;
+				let proxThresh = 10;
```
Comment (leeoniya) [inline src/uPlot.js:1883]: `snapProx`

File: src/uPlot.js [hunk 3]
```diff
@@ -1868,11 +1868,55 @@ export default function uPlot(opts, data, then) {
...
+
+				if (dragX && dragY) {
+					// maybe omni corner snap
+					xProx = mouseLeft1 <= proxThresh || mouseLeft1 >= plotWidCss - proxThresh;
+					yProx = mouseTop1  <= proxThresh || mouseTop1  >= plotHgtCss - proxThresh;
+				}
+				
+				if (dragX && xProx) {	
+					let xMin = min(dLft, dRgt);
+					if (xMin == dLft)
```
Comment (leeoniya) [inline src/uPlot.js:1893]: please make sure any leading `if ()` blocks have an empty line before them. more legible to me than having `let` immediately adjacent, especially with identical alignment.

**Comment by leeoniya** [comment]
> unless we want proxthresh to be a setting

let's wait till at least 1 person complains ;)

looks good. please address the tweaks, exclude the dist builds and squash into a better commit message:

```
snap selection to plot edges on mouseleave. close #244
```

**Comment by EmiPhil** [comment]
should be gtg

**Pull request merged**